### ISSUE TITLE
[question] the variance of multitaskGP prediction

### ISSUE THREAD

**Issue by zhaojingyao** [pytorch/botorch#526]
Hi, I used the MT model for prediction and found that the variance of the prediction does not reflect the uncertainty very well.

For example, when this graph is close to x=0, the prediction of the blue curve is obviously quite accurate, but the prediction variance is quite large. There seems to be a strong relationship between the variance of the prediction and the presence of samples in this region.

![image](https://user-images.githubusercontent.com/40418158/91298603-e804d280-e7d2-11ea-8431-dfb55d9c1bc3.png)

Here is my code:
   model = MultiTaskGP(train_X, train_Y, task_feature=-1)
   mll = ExactMarginalLogLikelihood(model.likelihood, model)
   fit_gpytorch_model(mll)

    posterior = model.posterior(test_x)
    mean = posterior.mean.cpu().detach() 
    var = posterior.variance.cpu().detach()
    std = torch.sqrt(var)
    mesh_y_func1 = mean[:,0]
    mesh_y_func2 = mean[:,1]
    std_func1 = std[:,0]
    std_func2 = std[:,1]

    plt.plot(mesh_x,mesh_y_func1,c='r', label='pre gain')
    plt.plot(mesh_x,mesh_y_func2,c='b', label='pre ugf')
    plt.fill_between(mesh_x, mesh_y_func1 + 1*std_func1, mesh_y_func1 - 1*std_func1, alpha=0.1, color="red")
    plt.fill_between(mesh_x, mesh_y_func2 + 1*std_func2, mesh_y_func2 - 1*std_func2, alpha=0.1, color="blue")
    plt.legend()
    plt.title('multi-task GP')

**Comment by Balandat** [comment]
This seems reasonable to me - the variance grows as you predict outside the support of the data - this is a common property of GPs (and generally what you want). Does this make sense?

**Comment by zhaojingyao** [comment]
> This seems reasonable to me - the variance grows as you predict outside the support of the data - this is a common property of GPs (and generally what you want). Does this make sense?

In my experiments, I found that curve fitting may become worse when more training points are given to multi-task-GP. This is also the case in the higher-dimensional case. Beyond that, in this case, the variance would be wildly off base, but the single-GP model would not be.

![image](https://user-images.githubusercontent.com/40418158/91691548-64fcc700-eb9a-11ea-8d6c-1576cb3fefe3.png)
![image](https://user-images.githubusercontent.com/40418158/91691559-6a5a1180-eb9a-11ea-843d-ea0cab0f3b0b.png)
                        
                        
 As the number of training points increases, does the hyperparameter training become more difficult? Is there an interface that allows users to manually adjust the number of iterations for parameter optimization?

**Comment by jrojsel** [comment]
@zhaojingyao these issues are most likely resolved by standardizing the data prior to model fitting. See [botorch.utils.transforms.standardize](https://botorch.org/api/utils.html#botorch.utils.transforms.standardize).

**Comment by Balandat** [comment]
assuming this has been resolved - feel free to reopen if necessary
